tsl/test/sql/continuous_aggs_variable_size_buckets.sql

```diff
@@ -491,3 +491,24 @@ DROP TABLE conditions_dist CASCADE;
 DROP DATABASE :DATA_NODE_1;
 DROP DATABASE :DATA_NODE_2;
 DROP DATABASE :DATA_NODE_3;
+
+-- Test the specific code path of creating a CAGG on top of empty hypertable.
+
+CREATE TABLE conditions_empty(
+  day DATE NOT NULL,
+  city text NOT NULL,
+  temperature INT NOT NULL);
+
+SELECT create_hypertable(
+  'conditions_empty', 'day',
+  chunk_time_interval => INTERVAL '1 day'
+);
+
+CREATE MATERIALIZED VIEW conditions_summary_empty
+WITH (timescaledb.continuous, timescaledb.materialized_only=true) AS
+SELECT city,
+   timescaledb_experimental.time_bucket_ng('1 month', day) AS bucket,
+   MIN(temperature),
+   MAX(temperature)
+FROM conditions_empty
+GROUP BY city, bucket;
```

Comment:
you should add a test that shows that refresh works after you add data to the table: 
1) add data to the hypertable.
2) call refresh_continuous_aggregate.
My primary concern is that this code changes the record that gets added to the invalidation log for an empty table and does this have any unintended side-effects?

Reply:
Thanks, @gayyappan I added the test 🙏